### PR TITLE
Add Aggregation, Insertion and Updating and Deleting statements

### PR DESCRIPTION
**Pull request by francksefu**
Add a column species of type string to your animals table. Modify your schema.sql file. -> done
Insert the following data: -> done
Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
Remember to add these insertions to your data.sql file.
Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
Inside a transaction: --> done
Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Verify that changes were made.
Commit the transaction.
Verify that changes persist after commit.
Take a screenshot of the results of your actions.
Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
Take a screenshot of the results of your actions.
Inside a transaction:
Delete all animals born after Jan 1st, 2022.
Create a savepoint for the transaction.
Update all animals' weight to be their weight multiplied by -1.
Rollback to the savepoint
Update all animals' weights that are negative to be their weight multiplied by -1.
Commit transaction
Take a screenshot of the results of your actions.
Remember to add all these transactions to your queries.sql file.
Write queries to answer the following questions:
How many animals are there?
How many animals have never tried to escape?
What is the average weight of animals?
Who escapes the most, neutered or not neutered animals?
What is the minimum and maximum weight of each type of animal?
What is the average number of escape attempts per animal type of those born between 1990 and 2000?
Take a screenshot of the results of your queries.
Remember to add these queries to your queries.sql file.
Remember to add screenshots with the results of your queries and with the results of your play with transactions to your Pull Request description. --> done

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -6,4 +6,47 @@ SELECT name from animals WHERE (neutered = TRUE)AND(espace_attempts < 3);
...
+  select * from animals;
+ROLLBACK;
+select * from animals;
+
+BEGIN;
+  UPDATE animals SET species ='digimon' WHERE name LIKE '%mon';
+  select * from animals;
+  UPDATE animals SET species ='pokemon' WHERE species IS NULL;
+  select * from animals;
+COMMIT;
```
Comment (addod19) [inline queries.sql:22]: - [ ] Kindly verify the changes persists after the `commit` as seen in the requirements
Screenshot from 2023-07-13 12-17-02

File: queries.sql [hunk 2]
```diff
@@ -6,4 +6,47 @@ SELECT name from animals WHERE (neutered = TRUE)AND(espace_attempts < 3);
...
+
+BEGIN;
+  DELETE FROM animals;
+ROLLBACK;
+select * from animals;
+
+BEGIN;
+  DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+  select * from animals;
```
Comment (addod19) [inline queries.sql:32]: - [ ] After deleting all animals born after Jan 1st, 2022, we have to create a savepoint for the transaction as seen in the requirements
Screenshot from 2023-07-13 12-20-40

File: queries.sql [hunk 3]
```diff
@@ -6,4 +6,47 @@ SELECT name from animals WHERE (neutered = TRUE)AND(espace_attempts < 3);
...
+COMMIT;
+
+ SELECT COUNT(*) FROM animals;
+ SELECT COUNT(*) FROM animals WHERE espace_attempts = 0;
+ SELECT AVG(weight_kg) FROM animals;
+ SELECT neutered, MAX(espace_attempts) FROM animals GROUP BY neutered;
+ select MIN(weight_kg), MAX(weight_kg), species from animals GROUP BY species;
+
+ --for the last
+ SELECT AVG(espace_attempts), species FROM (SELECT*FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-12') as animal GROUP BY species;
```
Comment (addod19) [inline queries.sql]: - [ ] Kindly change `BETWEEN '1990-01-01' AND '2000-12-12')` to `BETWEEN '1990-01-01' AND '2000-12-31')` in order to fulfill the requirements below
Screenshot from 2023-07-13 12-25-21

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @francksefu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [ ] Correct Git-flow followed :+1:
- [ ] Nice attempt for a start :100: 

## Required Changes ♻️
_Check the comments under the review._
- [ ] It would be better if we add a column `species` to our Animal table as seen in the project requirements. It is quiet strange you don't have a column `species` in the table but the screenshot of the query results does
Screenshot from 2023-07-13 12-03-05
- [ ] It would be better if we add the screenshots to our PR descriptions
- [ ] It would be better if we complete this section of the README file, i can not understand what you mean. Fixing it will make our README more professional
Screenshot from 2023-07-13 12-29-54
- [ ] Kindly make the appropriate changes, since this is not a capstone, we should have the current project instead
Screenshot from 2023-07-13 12-32-29
- [ ] Kindly remove these from the README file. These are placeholders text that needs to be removed
Screenshot from 2023-07-13 12-32-06
Screenshot from 2023-07-13 12-33-49



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by francksefu** [comment]
Thanks for the review, I implement all of them, and the screenshot it's in my repository as the requirements said

**Comment by t-yanick** [comment]
- [ ] Great job so far. Your screenshots should be in your Pull request body and not in your repository

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @francksefu , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All requested changes implemented. :+1: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Adding your query results screenshots to your pull request description before merging.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**